Refactoring plus new update

lucival1 opened the pull request.

Samox commented on pharmacy.test.js:
```diff
@@ -1,9 +1,120 @@
     );
   });
+
+  it("should not decrease the benefit below 0", () => {
```
what about the benefit evolution after the expiry date ?

Samox commented on pharmacy.js:
```diff
@@ -1,65 +1,91 @@
-      }
-    }
+    this.drugs.forEach(drug => drug.update());
+    // for (var i = 0; i < this.drugs.length; i++) {
```
at this point your tests are not green

Samox commented on pharmacy.js:
```diff
@@ -24,10 +24,25 @@ export class Drug {
+      // update benefit according to days left before expiration
+      drugEffects.benefitIncrementsByDays.forEach(incrementByDays => {
+        if(this.expiresIn <= incrementByDays.days && processed === false) {
+          processed = true;
```
code smell

lucival1 committed:
Add Abstract DrugModel
Refactor Folder structure
Add tests for DrugModel and Doliprane classes

lucival1 committed:
Add Herbal Tea class
Add tests for Herbal Tea classes
Add maxBenefitValue to DrugModel

lucival1 committed:
Add Fervex class
Add tests for Fervex classes

lucival1 committed:
Add Magic Pill class
Add test for Magic Pill class

lucival1 committed:
Move Drugs into App folder
Improve DrugModel updateBenefitValue - benefit decrease twice as fast when expired
Improve tests for DrugModel updateBenefitValue

lucival1 committed:
Move Pharmacy into App folder
Move Pharmacy.test into test folder and add more tests
Improve index

lucival1 committed:
Remove canProcess method from Drug classes
Clean up comments and code not used
Fix Schema test on DrugModel

lucival1 committed:
Add Dafalgan Class and test
Final clean up

lucival1 committed:
Refactor drug classes and test into strategy pattern
Create DrugStrategy Interface
Reorganize directories
Fix Api Client
Add extra Pharmacy tests
